pkg/appgw/configbuilder.go

```diff
@@ -182,19 +183,37 @@ func generateBackendID(ingress *v1beta1.Ingress, rule *v1beta1.IngressRule, path
 	}
 }
 
-func generateListenerID(rule *v1beta1.IngressRule, protocol n.ApplicationGatewayProtocol, overridePort *Port, usePrivateIP bool) listenerIdentifier {
+func generateListenerID(ingress *v1beta1.Ingress, rule *v1beta1.IngressRule, protocol n.ApplicationGatewayProtocol, overridePort *Port, usePrivateIP bool) listenerIdentifier {
 	frontendPort := Port(80)
 	if protocol == n.HTTPS {
 		frontendPort = Port(443)
 	}
 	if overridePort != nil {
 		frontendPort = *overridePort
 	}
+
 	listenerID := listenerIdentifier{
 		FrontendPort: frontendPort,
-		HostName:     rule.Host,
 		UsePrivateIP: usePrivateIP,
 	}
+
+	// Populate the hostnames property
+	var hostnames []string
+	if rule.Host != "" {
+		hostnames = append(hostnames, rule.Host)
+	}
+
+	if extendedHostNames, err := annotations.GetHostNameExtensions(ingress, rule.Host); err == nil {
+		if extendedHostNames != nil {
+			hostnames = append(hostnames, extendedHostNames...)
+		}
+	} else {
+		glog.Infof("GetHostNameExtensions has error: %s", err)
```

Comment:
Lets log this at `V(5)`. Also, if the error is `ErrMissingAnnotations`, we should skip logging. https://github.com/Azure/application-gateway-kubernetes-ingress/blob/2bc0e405eb9efe5e9eb45a2e00953e4b9de72039/pkg/annotations/errors.go#L25

Reply:
I have removed this log line. It was spitting out too many times. So completely removed it